### PR TITLE
Completed dao8

### PR DESCRIPTION
**Pull request by Vedroid**

### CODE REVIEW

**Review by Resci — APPROVED** [review]

File: src/main/java/ua/vedroid/dao3/controllers/DeleteDataController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+            throws ServletException, IOException {
+        String query = "TRUNCATE TABLE taxi_service.manufacturers RESTART IDENTITY CASCADE; "
+                + "TRUNCATE TABLE taxi_service.cars RESTART IDENTITY CASCADE; "
+                + "TRUNCATE TABLE taxi_service.drivers RESTART IDENTITY CASCADE; ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.executeUpdate();
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Truncate tables is failed", ex);
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/DeleteDataController.java]: you definitely should not have SQL queries in controllers and I don't think that you should have such functionality accessible via an endpoint.  Imagine such an endpoint in the real project: 
https://rozetka.com.ua/ua/computers-notebooks/clear

File: src/main/java/ua/vedroid/dao3/controllers/car/AddDriversController.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            if (car == null) {
+                req.setAttribute("msg", "Car with id=" + carId
+                        + " not found!");
+            } else if (driver == null) {
+                req.setAttribute("msg", "Driver with id=" + driverId
+                        + " not found!");
+            } else {
+                req.setAttribute("msg", "Failed to add driver(id=" + driver.getId()
+                        + ") to car(id=" + car.getId() + ")!");
+            }
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/car/AddDriversController.java]: let's just pass the whole exception as attributes and remove these conditions

File: src/main/java/ua/vedroid/dao3/controllers/driver/DeleteDriverController.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+public class DeleteDriverController extends HttpServlet {
+    private static final Injector injector =
+            Injector.getInstance(ApplicationStarter.class.getPackageName());
+    private DriverService service =
+            (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        service.deleteById(Long.valueOf(req.getParameter("id")));
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/driver/DeleteDriverController.java]:
```suggestion
        service.deleteById(Long.valueOf(req.getParameter("driver_id")));
```

File: src/main/java/ua/vedroid/dao3/controllers/manufacturer/CreateManufacturerController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher(REGISTRATION_JSP).forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        String name = req.getParameter("name");
+        String country = req.getParameter("country");
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/manufacturer/CreateManufacturerController.java]:
```suggestion
        String name = req.getParameter("manufacturer_name");
        String country = req.getParameter("manufacturer_country");
```

File: src/main/java/ua/vedroid/dao3/controllers/manufacturer/DeleteManufacturerController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+public class DeleteManufacturerController extends HttpServlet {
+    private static final Injector injector =
+            Injector.getInstance(ApplicationStarter.class.getPackageName());
+    private ManufacturerService service =
+            (ManufacturerService) injector.getInstance(ManufacturerService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long manufacturerId = Long.valueOf(req.getParameter("id"));
```
Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/manufacturer/DeleteManufacturerController.java]: add model names to all parameters in all controllers

Comment (dmytro-hryhoruk) [inline src/main/java/ua/vedroid/dao3/controllers/manufacturer/DeleteManufacturerController.java]:
```suggestion
        Long manufacturerId = Long.valueOf(req.getParameter("manufacturer_id"));
```

File: src/main/webapp/WEB-INF/views/cars/all.jsp [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        <h1>All cars page</h1>
+        <a href="${pageContext.request.contextPath}/">Go to main page</a>
+        <table>
+            <tr>
+                <th>ID</th>
+                <th>Model</th>
+                <th>Manufacturer</th>
+                <th>Drivers</th>
+            </tr>
+            <%--@elvariable id="cars" type="java.util.List"--%>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/cars/all.jsp]:
```suggestion
```

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 7]
```diff
@@ -1,9 +1,62 @@
-<%@ page contentType="text/html;charset=UTF-8" language="java" %>
+<jsp:useBean id="msg" scope="request" type="java.lang.String"/>
+<%@ page contentType="text/html;charset=UTF-8" %>
 <html>
     <head>
         <title>Title</title>
     </head>
     <body>
         <h1>Hello world!</h1>
+
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/WEB-INF/views/index.jsp]:
```suggestion
```

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -10,6 +10,117 @@
...
+        <url-pattern>/</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>injectData</servlet-name>
+        <servlet-class>ua.vedroid.dao3.controllers.InjectDataController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>injectData</servlet-name>
+        <url-pattern>/injectData</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/webapp/web.xml]:
```suggestion
        <url-pattern>/inject</url-pattern>
```

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]
Good job

File: src/main/java/ua/vedroid/dao3/controllers/car/AddDriversController.java (hunk 2 above)
Comment (Vedroid) [reply src/main/java/ua/vedroid/dao3/controllers/car/AddDriversController.java]: @dmytro-hryhoruk
Так?
```
catch (NoSuchElementException ex) {
    req.setAttribute("msg", ex);
    req.getRequestDispatcher(ADD_DRIVER_JSP).forward(req, resp);
}
```
image

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**